src/cloud_provider/digitalocean/common.rs

```diff
@@ -23,21 +28,41 @@ pub fn kubernetes_config_path(
         workspace_directory, kubernetes_cluster_id
     );
 
-    let kubeconfig = download_space_object(
-        spaces_access_id,
-        spaces_secret_key,
-        kubernetes_config_bucket_name.as_str(),
-        kubernetes_config_object_key.as_str(),
-        region,
-    );
-    match kubeconfig {
-        Ok(body) => {
+    let result = retry::retry(Fixed::from_millis(3000).take(5), || {
+        let try_kubeconfig = download_space_object(
+            spaces_access_id,
+            spaces_secret_key,
+            kubernetes_config_bucket_name.as_str(),
+            kubernetes_config_object_key.as_str(),
+            region,
+        );
+        match try_kubeconfig {
+            Ok(kubeconfig) => OperationResult::Ok(kubeconfig),
+            Err(err) => {
+                warn!("Failed to download the kubeconfig file, retrying...");
+                return OperationResult::Err(format!(
+                    "Unable to download the kubeconfig file from space: {:?}",
+                    err
+                ));
+            }
+        }
+    });
+
+    match result {
+        Ok(downloaded) => {
             let mut file =
                 File::create(kubernetes_config_file_path.clone()).expect("unable to create file");
-            file.write_all(body.as_bytes()).expect("unable to write");
+            file.write_all(downloaded.as_bytes())
+                .expect("unable to write");
```

Comment:
can you please give more context on the message, it would be easier for debug purpose

Reply:
done add some comment upper ;)